shop/src/main/java/ro/msg/learning/shop/controller/CustomerController.java

```diff
@@ -0,0 +1,54 @@
+package ro.msg.learning.shop.controller;
+
+import org.springframework.web.bind.annotation.*;
+import ro.msg.learning.shop.model.Customer;
+import ro.msg.learning.shop.payroll.CustomerNotFoundException;
+import ro.msg.learning.shop.repository.ICustomerRepository;
+
+import java.util.List;
+
+@RestController
+public class CustomerController {
+    private final ICustomerRepository repository;
+
+    CustomerController(ICustomerRepository repository){
+        this.repository=repository;
+    }
+
+    @GetMapping("/customers")
+    List<Customer> all() {
+        return repository.findAll();
+    }
+
+    @PostMapping("/customers")
+    Customer newCustomer(@RequestBody Customer newCustomer) {
+        return repository.save(newCustomer);
+    }
+
+    @GetMapping("/customers/{id}")
+    Customer one(@PathVariable Integer id) {
+
+        return repository.findById(id)
+                .orElseThrow(() -> new CustomerNotFoundException(id));
```

Comment:
This is again logic that needs to be handled within a service. The service is the business logic responsible